### PR TITLE
[#146757471] Support TM search

### PR DESCRIPTION
**Pull request by toshipon**
support endpoint https://wiki.memsource.com/wiki/Translation_Memory_API_v4#Search

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 __pycache__
```
Comment (kelvintaywl) [inline .gitignore]: maybe you can use the [suggested Python .gitignore file as seen here](https://github.com/github/gitignore/blob/master/Python.gitignore)

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
...
 .coverage
+*.pyc
```
Comment (kelvintaywl) [inline .gitignore]: maybe you can use the [suggested Python .gitignore file as seen here](https://github.com/github/gitignore/blob/master/Python.gitignore)

File: test/api/test_translation_memory.py [hunk 3]
```diff
@@ -174,6 +174,106 @@ def test_upload_from_text(self, mock_request):
         self.assertEqual(accepted_segments_count, returned_value)
 
     @patch.object(requests.Session, 'request')
+    def test_search(self, mock_request):
+        type(mock_request()).status_code = PropertyMock(return_value=200)
+
+        translation_memory_id = 1234
+        query = "Hello"
+        source_lang = "en"
+        target_langs = ["ja", 'en-gb']
```
Comment (narusemotoki) [inline test/api/test_translation_memory.py]: I think `en_gb` in memsource.

File: test/api/test_translation_memory.py [hunk 4]
```diff
@@ -174,6 +174,106 @@ def test_upload_from_text(self, mock_request):
...
+                'query': query,
+                'sourceLang': source_lang,
+                'targetLang': target_langs,
+                'nextSegment': next_segment,
+                'previousSegment': previous_segment
+            },
+            timeout=constants.Base.timeout.value
+        )
+
+        for segment_search_result in returned_value:
```
Comment (narusemotoki) [inline test/api/test_translation_memory.py]: I think you need to assert returned_value is not 0 length.

**Pull request merged**